wallet-v2 recover panics if you ask it to create 1 account

#🐞 Bug Report

Create a new wallet using `wallet-v2 recover` and when asked "Enter how many accounts you would like to recover...." Enter 1.  This will lead to a panic:

```
[2020-09-26 00:23:43] ERROR main: Runtime panic: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0xc00173f608, 0x14f4000, 0x2503330)
	GOROOT/src/runtime/debug/stack.go:24 +0x9d
main.main.func9()
	validator/main.go:345 +0x66
panic(0x14f4000, 0x2503330)
	GOROOT/src/runtime/panic.go:969 +0x166
github.com/prysmaticlabs/prysm/validator/accounts/v2/wallet.(*Wallet).SaveHashedPassword(0x0, 0x1a7b140, 0xc000120000, 0x0, 0x0)
	validator/accounts/v2/wallet/wallet.go:411 +0x37
github.com/prysmaticlabs/prysm/validator/accounts/v2.RecoverWalletCli(0xc00020d500, 0xc000fcf9b8, 0x8)
	validator/accounts/v2/wallet_recover.go:75 +0x289
github.com/prysmaticlabs/prysm/validator/accounts/v2.glob..func10(0xc00020d500, 0x9, 0x10)
	validator/accounts/v2/cmd_wallet.go:77 +0x3d
github.com/urfave/cli/v2.(*Command).Run(0x251c4c0, 0xc00020d240, 0x0, 0x0)
	external/com_github_urfave_cli_v2/command.go:164 +0x4e0
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000305680, 0xc00020d100, 0x0, 0x0)
	external/com_github_urfave_cli_v2/app.go:427 +0xa68
github.com/urfave/cli/v2.(*Command).startApp(0x251c160, 0xc00020d100, 0x7ffff399b715, 0x9)
	external/com_github_urfave_cli_v2/command.go:279 +0x68c
github.com/urfave/cli/v2.(*Command).Run(0x251c160, 0xc00020d100, 0x0, 0x0)
	external/com_github_urfave_cli_v2/command.go:94 +0xa12
github.com/urfave/cli/v2.(*App).RunContext(0xc000305380, 0x1a7b140, 0xc000120000, 0xc00011c150, 0x3, 0x3, 0x0, 0x0)
	external/com_github_urfave_cli_v2/app.go:306 +0x814
github.com/urfave/cli/v2.(*App).Run(...)
	external/com_github_urfave_cli_v2/app.go:215
main.main()
	validator/main.go:350 +0x91f

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x109ac97]

goroutine 1 [running]:
main.main.func9()
	validator/main.go:346 +0x13e
panic(0x14f4000, 0x2503330)
	GOROOT/src/runtime/panic.go:969 +0x166
github.com/prysmaticlabs/prysm/validator/accounts/v2/wallet.(*Wallet).SaveHashedPassword(0x0, 0x1a7b140, 0xc000120000, 0x0, 0x0)
	validator/accounts/v2/wallet/wallet.go:411 +0x37
github.com/prysmaticlabs/prysm/validator/accounts/v2.RecoverWalletCli(0xc00020d500, 0xc000fcf9b8, 0x8)
	validator/accounts/v2/wallet_recover.go:75 +0x289
github.com/prysmaticlabs/prysm/validator/accounts/v2.glob..func10(0xc00020d500, 0x9, 0x10)
	validator/accounts/v2/cmd_wallet.go:77 +0x3d
github.com/urfave/cli/v2.(*Command).Run(0x251c4c0, 0xc00020d240, 0x0, 0x0)
	external/com_github_urfave_cli_v2/command.go:164 +0x4e0
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000305680, 0xc00020d100, 0x0, 0x0)
	external/com_github_urfave_cli_v2/app.go:427 +0xa68
github.com/urfave/cli/v2.(*Command).startApp(0x251c160, 0xc00020d100, 0x7ffff399b715, 0x9)
	external/com_github_urfave_cli_v2/command.go:279 +0x68c
github.com/urfave/cli/v2.(*Command).Run(0x251c160, 0xc00020d100, 0x0, 0x0)
	external/com_github_urfave_cli_v2/command.go:94 +0xa12
github.com/urfave/cli/v2.(*App).RunContext(0xc000305380, 0x1a7b140, 0xc000120000, 0xc00011c150, 0x3, 0x3, 0x0, 0x0)
	external/com_github_urfave_cli_v2/app.go:306 +0x814
github.com/urfave/cli/v2.(*App).Run(...)
	external/com_github_urfave_cli_v2/app.go:215
main.main()
	validator/main.go:350 +0x91f
runtime: note: your Linux kernel may be buggy
runtime: note: see https://golang.org/wiki/LinuxKernelSignalVectorBug
runtime: note: mlock workaround for kernel bug failed with errno 12
```


### Description

See above

### Has this worked before in a previous version?

I don't really know.


## 🔬 Minimal Reproduction

See above

## 🌍  Your Environment

**Operating System:**

Ubuntu VM running Docker

**What version of Prysm are you running? (Which release)**

Current master.  282f3eec01470935e65460163755fa00595c1ce8

**Anything else relevant (validator index / public key)?**
No

--- a/validator/accounts/v2/wallet_recover.go
+++ b/validator/accounts/v2/wallet_recover.go
@@ -114,7 +114,7 @@ func RecoverWallet(ctx context.Context, cfg *RecoverWalletConfig) (*wallet.Walle
 		if _, err := km.CreateAccount(ctx, true /*logAccountInfo*/); err != nil {
 			return nil, errors.Wrap(err, "could not create account in wallet")
 		}
-		return nil, nil
+		return w, nil
 	}
 	for i := int64(0); i < cfg.NumAccounts; i++ {
 		if _, err := km.CreateAccount(ctx, false /*logAccountInfo*/); err != nil {

--- a/validator/accounts/v2/wallet_recover_test.go
+++ b/validator/accounts/v2/wallet_recover_test.go
@@ -81,3 +81,42 @@ func TestRecoverDerivedWallet(t *testing.T) {
 	require.Equal(t, len(names), int(numAccounts))
 
 }
+
+// TestRecoverDerivedWallet_OneAccount is a test for regression in cases where the number of accounts recovered is 1
+func TestRecoverDerivedWallet_OneAccount(t *testing.T) {
+	testDir := testutil.TempDir()
+	walletDir := filepath.Join(testDir, walletDirName)
+	exportDir := filepath.Join(testDir, exportDirName)
+	defer func() {
+		assert.NoError(t, os.RemoveAll(walletDir))
+		assert.NoError(t, os.RemoveAll(exportDir))
+	}()
+
+	passwordFilePath := filepath.Join(testDir, passwordFileName)
+	require.NoError(t, ioutil.WriteFile(passwordFilePath, []byte(password), os.ModePerm))
+	mnemonicFilePath := filepath.Join(testDir, mnemonicFileName)
+	require.NoError(t, ioutil.WriteFile(mnemonicFilePath, []byte(mnemonic), os.ModePerm))
+
+	numAccounts := int64(1)
+	app := cli.App{}
+	set := flag.NewFlagSet("test", 0)
+	set.String(flags.WalletDirFlag.Name, walletDir, "")
+	set.String(flags.WalletPasswordFileFlag.Name, passwordFilePath, "")
+	set.String(flags.KeymanagerKindFlag.Name, v2keymanager.Derived.String(), "")
+	set.String(flags.MnemonicFileFlag.Name, mnemonicFilePath, "")
+	set.Int64(flags.NumAccountsFlag.Name, numAccounts, "")
+	assert.NoError(t, set.Set(flags.WalletDirFlag.Name, walletDir))
+	assert.NoError(t, set.Set(flags.WalletPasswordFileFlag.Name, passwordFilePath))
+	assert.NoError(t, set.Set(flags.KeymanagerKindFlag.Name, v2keymanager.Derived.String()))
+	assert.NoError(t, set.Set(flags.MnemonicFileFlag.Name, mnemonicFilePath))
+	assert.NoError(t, set.Set(flags.NumAccountsFlag.Name, strconv.Itoa(int(numAccounts))))
+	cliCtx := cli.NewContext(&app, set, nil)
+
+	require.NoError(t, RecoverWalletCli(cliCtx))
+
+	_, err := wallet.OpenWallet(cliCtx.Context, &wallet.Config{
+		WalletDir:      walletDir,
+		WalletPassword: password,
+	})
+	assert.NoError(t, err)
+}

--- a/validator/accounts/v2/wallet_recover_test.go
+++ b/validator/accounts/v2/wallet_recover_test.go
@@ -86,10 +86,8 @@ func TestRecoverDerivedWallet(t *testing.T) {
 func TestRecoverDerivedWallet_OneAccount(t *testing.T) {
 	testDir := testutil.TempDir()
 	walletDir := filepath.Join(testDir, walletDirName)
-	exportDir := filepath.Join(testDir, exportDirName)
 	defer func() {
 		assert.NoError(t, os.RemoveAll(walletDir))
-		assert.NoError(t, os.RemoveAll(exportDir))
 	}()
 
 	passwordFilePath := filepath.Join(testDir, passwordFileName)

